ilc/client/GuardManager.js

```diff
@@ -0,0 +1,112 @@
+const navigateToUrl = require('single-spa').navigateToUrl;
+const unlocalizeUrl = require('../common/i18n/unlocalizeUrl');
+
+const actionTypes = require('../common/guard/actionTypes');
+
+const getIlcRouter = require('./getIlcRouter');
+const getIlcConfig = require('./ilcConfig');
+
+class GuardManager {
+    #registryConf;
+    #router;
+    #unlocalizeUrl;
+    #navigateToUrl;
+    #transitionHooksPlugin;
+
+    constructor(
+        registryConf,
+        router,
+        unlocalizeUrl,
+        navigateToUrl,
+        transitionHooksPlugin,
+    ) {
+        this.#registryConf = registryConf;
+        this.#router = router;
+        this.#unlocalizeUrl = unlocalizeUrl;
+        this.#navigateToUrl = navigateToUrl;
+        this.#transitionHooksPlugin = transitionHooksPlugin;
+    }
+
+    shouldNavigateTo(url) {
+        const pathname = new URL(url, 'http://hack').pathname;
+        const route = this.#router.match(this.#unlocalizeUrl(this.#registryConf.settings.i18n, pathname));
+        const hooks = this.#transitionHooksPlugin.getTransitionHooks();
+
+        if (route.specialRole == null) {
+            for (const hook of hooks) {
+                const action = hook({
+                    route: {
+                        ...route,
+                        url,
+                    },
+                    navigate: this.#navigateToUrl,
+                });
+
+                switch(action.type) {
+                    case actionTypes.stopNavigation: return false;
+                    case actionTypes.continue:
+                    default: break;
+                }
+            }
+        }
+
+        return true;
+    }
+};
+
+let guardManager;
+
+module.exports = () => {
+    if (guardManager === undefined) {
+        const router = getIlcRouter();
+        const registryConf = getIlcConfig();
+        const transitionHooksPlugin = {
```

Comment:
Mocked `transitionHooks` plugin

Reply:
Pls move it to separate file & load via some PluginManager, as we do it at server side

Reply:
still relevant

Reply:
Moved to the folder `ilc-plugin-transition-hooks` that we can move to `node_modules` into any folder (e.g. `node_modules/@test/ilc-plugin-transition-hooks`) for testing